pkg/cache/cache.go

```diff
@@ -118,44 +119,59 @@ type Options struct {
 	// Mapper is the RESTMapper to use for mapping GroupVersionKinds to Resources
 	Mapper meta.RESTMapper
 
-	// Resync is the base frequency the informers are resynced.
+	// ResyncEvery is the base frequency the informers are resynced.
 	// Defaults to defaultResyncTime.
-	// A 10 percent jitter will be added to the Resync period between informers
+	// A 10 percent jitter will be added to the ResyncEvery period between informers
 	// So that all informers will not send list requests simultaneously.
-	Resync *time.Duration
+	ResyncEvery *time.Duration
 
-	// Namespace restricts the cache's ListWatch to the desired namespace
-	// Default watches all namespaces
-	Namespace string
+	// View restricts the cache's ListWatch to the desired fields per GVK
+	// Default watches all fields and all namespaces.
+	View ViewOptions
+}
 
-	// SelectorsByObject restricts the cache's ListWatch to the desired
-	// fields per GVK at the specified object, the map's value must implement
-	// Selector [1] using for example a Set [2]
-	// [1] https://pkg.go.dev/k8s.io/apimachinery/pkg/fields#Selector
-	// [2] https://pkg.go.dev/k8s.io/apimachinery/pkg/fields#Set
-	SelectorsByObject SelectorsByObject
+// ViewOptions are the optional arguments for creating a cache view.
+// A cache view restricts the Get(), List(), and internal ListWatch operations
+// to the desired parameter.
+type ViewOptions struct {
+	// Namespaces restricts the cache's ListWatch to the desired namespaces
+	// Default watches all namespaces
+	Namespaces []string
 
 	// DefaultSelector will be used as selectors for all object types
-	// that do not have a selector in SelectorsByObject defined.
+	// unless they have a more specific selector set in ByObject.
 	DefaultSelector ObjectSelector
 
-	// UnsafeDisableDeepCopyByObject indicates not to deep copy objects during get or
-	// list objects per GVK at the specified object.
-	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
-	// otherwise you will mutate the object in the cache.
-	UnsafeDisableDeepCopyByObject DisableDeepCopyByObject
+	// DefaultTransform will be used as transform for all object types
+	// unless they have a more specific transform set in ByObject.
+	DefaultTransform toolscache.TransformFunc
+
+	// ByObject restricts the cache's ListWatch to the desired fields per GVK at the specified object.
+	ByObject ViewByObject
```

Comment:
I don't really understand how it improves anything to move this into a substruct, IMHO this just makes it harder to find what options exist.

What might have improved things would be if this was the object -> object-level options map rather than a struct that embedds a bunch of maps.